configuration/steps/remote.py

```diff
@@ -21,24 +23,34 @@ def __init__(
         options: StepOptions = None,
         interrupt_signal="TERM",
         env_vars: list[tuple] = None,
+        url=None,
```

Comment:
Let's make this url a dataclass.

@dataclass
class URL:
    url: str
    url_text: str = None
    
We only keep one parameter to ShellStep.